### PR TITLE
Update S.R.Metadata in HostModel

### PR DESCRIPTION
**Pull request by am11**
Use shared version of `System.Reflection.Metadata` in HostModel. It is compatible with netstandard2.0.

### CODE REVIEW

File: eng/Versions.props [hunk 1]
```diff
@@ -74,8 +74,6 @@
     <NuGetBuildTasksPackVersion>6.0.0-preview.1.102</NuGetBuildTasksPackVersion>
     <!-- Installer dependencies -->
     <MicrosoftNETCoreAppRuntimewinx64Version>7.0.0-rc.1.22414.6</MicrosoftNETCoreAppRuntimewinx64Version>
-    <MicrosoftNETCoreDotNetHostVersion>7.0.0-rc.1.22414.6</MicrosoftNETCoreDotNetHostVersion>
-    <MicrosoftNETCoreDotNetHostPolicyVersion>7.0.0-rc.1.22414.6</MicrosoftNETCoreDotNetHostPolicyVersion>
```
Comment (am11) [inline eng/Versions.props:78]: These are unused and removed by https://github.com/dotnet/runtime/pull/73095, but inadvertently undone by https://github.com/dotnet/runtime/pull/72934.

File: src/installer/managed/Microsoft.NET.HostModel/Microsoft.NET.HostModel.csproj [hunk 2]
```diff
@@ -19,7 +19,7 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="System.Reflection.Metadata" Version="1.8.0" />
+    <PackageReference Include="System.Reflection.Metadata" Version="$(SystemReflectionMetadataVersion)" />
```
Comment (jkoritzinsky) [inline src/installer/managed/Microsoft.NET.HostModel/Microsoft.NET.HostModel.csproj:22]: We need to ensure that we're matching the version of `System.Reflection.Metadata` that is in the SDK, as version mismatches can cause significant issues when running on the Full Framework version of MSBuild.

Comment (jkoritzinsky) [reply src/installer/managed/Microsoft.NET.HostModel/Microsoft.NET.HostModel.csproj:22]: Looks like we can update to 6.0.0 (the current value of `SystemReflectionMetadataVersion`) as it's the same as the sdk's version.

**Review by jkoritzinsky — APPROVED** [review]

File: src/installer/managed/Microsoft.NET.HostModel/Microsoft.NET.HostModel.csproj (hunk 2 above)
Comment (am11) [reply src/installer/managed/Microsoft.NET.HostModel/Microsoft.NET.HostModel.csproj:22]: Yup, I first tested updating `SystemReflectionMetadataVersion` to 7.0 preview7, but `ILCompiler.Build.Tasks.csproj` requirement is that we keep `SystemCollectionsImmutableVersion` and `SystemReflectionMetadataVersion` in sync. Updating `System.Collections.Immutable` to 7.0 preview7  was failing Pri0 tests (`XUnitWrapperGenerator` was unable to load S.C.I among other things), so I reverted that change.

**Pull request merged**